### PR TITLE
[CHANGE] new parsing of WWW-Authenticate header

### PR DESCRIPTION
**Pull request by yTI0r**
Fix of #66
* New regexp and simplified parsing of WWW-Authenticate header

### CODE REVIEW

**Comment by TecDroiD** [comment]
After pulling your PR I get the following error

Python 3.10.7 (main, Sep  8 2022, 14:34:29) [GCC 12.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from pyVoIP.VoIP import VoIPPhone, InvalidStateError
>>> ph = VoIPPhone("fritz.box", 5060, "tarantulla", "password")
>>> ph.start()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/tecdroid/trash/pyVoIP/pyVoIP/VoIP.py", line 656, in start
    self.sip.start()
  File "/home/tecdroid/trash/pyVoIP/pyVoIP/SIP.py", line 946, in start
    self.register()
  File "/home/tecdroid/trash/pyVoIP/pyVoIP/SIP.py", line 1716, in register
    regRequest = self.genRegister(response)
  File "/home/tecdroid/trash/pyVoIP/pyVoIP/SIP.py", line 1196, in genRegister
    return self.gen_register(request, deregister)
  File "/home/tecdroid/trash/pyVoIP/pyVoIP/SIP.py", line 1199, in gen_register
    response = str(self.genAuthorization(request), "utf8")
  File "/home/tecdroid/trash/pyVoIP/pyVoIP/SIP.py", line 1055, in genAuthorization
    return self.gen_authorization(request)
  File "/home/tecdroid/trash/pyVoIP/pyVoIP/SIP.py", line 1058, in gen_authorization
    realm = request.authentication["realm"]
KeyError: 'realm'

Seems that your modification loses the realm-entry somehow

**Comment by tayler6000** [comment]
> Seems that your modification loses the realm-entry somehow

Just to clarify, did your code work before this PR? @TecDroiD

**Comment by tayler6000** [comment]
@yTI0r Please point this towards master, not development.

**Comment by TecDroiD** [comment]
@tayler6000  Sadly not, I had issue #65 where I also described what i've found out.

**Comment by tayler6000** [comment]
@TecDroiD looking at the examples you put I can see why both of these regex's don't fix your problem.  If you're ok installing from source I can push a fix to master that will fix your issue in the interim until this PR is fixed and merged.  Just @ me in your reply if you'd like that.

File: pyVoIP/SIP.py [hunk 1]
```diff
@@ -341,7 +341,7 @@ def __init__(self, data: bytes):
         self.body: Dict[str, Any] = {}
         self.authentication: Dict[str, str] = {}
         self.raw = data
-        self.auth_match = re.compile(",?[a-zA-Z0-9]*=")
+        self.auth_match = re.compile('(\w+)=("\w+"|\w+)')
```
Comment (tayler6000) [inline pyVoIP/SIP.py]: For `("\w+"|\w+)`, unfortunately `\w` does not cover `.` or other printable characters, which makes this solution not work for @TecDroiD 

Is there a better meta character that's closer to `string.printable`? Or would we need to do something like `[\w.,:;@]+` (obviously more symbols than that would be necessary but for examples sake).

**Comment by yTI0r** [comment]
In case @TecDroiD - dot in text of realm don't parsed correctly. So i inspired by the text of RFC2617 committed new regex.
@tayler6000 i changed the base branch from development to master as you said.

**Review by tayler6000 — APPROVED** [review]

**Comment by tayler6000** [comment]
This PR will be pushed as 1.6.1 within the next 24 hours.

**Pull request merged**